assets/app/scripts/directives/deleteLink.js

```diff
@@ -46,41 +116,42 @@ angular.module("openshiftConsole")
           });
 
           modalInstance.result.then(function() {
-          // upon clicking delete button, delete resource and send alert
+            // upon clicking delete button, delete resource and send alert
             var kind = scope.kind;
             var resourceName = scope.resourceName;
-            var projectName = scope.projectName;
             var typeDisplayName = scope.typeDisplayName || $filter('humanizeKind')(kind);
             var formattedResource = typeDisplayName + ' ' + "\'"  + (scope.displayName ? scope.displayName : resourceName) + "\'";
             var context = (scope.kind === 'Project') ? {} : {namespace: scope.projectName};
 
-            DataService.delete(APIService.kindToResource(kind), resourceName, context)
+            DataService.delete({
+              resource: APIService.kindToResource(kind),
+              // group or undefined
+              group: scope.group
+            }, resourceName, context)
             .then(function() {
-              if (kind !== 'Project') {
-                AlertMessageService.addAlert({
-                  name: resourceName,
-                  data: {
-                    type: "success",
-                    message: formattedResource + " was marked for deletion."
-                  }
+              showAlert({
+                name: resourceName,
+                data: {
+                  type: "success",
+                  message: formattedResource + " was marked for deletion."
+                }
+              });
+
+              // Delete any associated HPAs if requested.
+              var promises = [];
+              if (scope.options.deleteHPAs) {
+                _.forEach(scope.hpaList, function(hpa) {
+                  promises.push(deleteHPA(hpa));
                 });
-                Navigate.toResourceList(APIService.kindToResource(kind), projectName);
               }
-              else {
-                if ($location.path() === '/') {
-                  scope.$emit('deleteProject');
-                }
-                else if ($location.path().indexOf('settings') > '-1') {
-                  var homeRedirect = URI('/');
-                  AlertMessageService.addAlert({
-                    name: resourceName,
-                    data: {
-                      type: "success",
-                      message: formattedResource + " was marked for deletion."
-                    }
-                  });
-                  $location.url(homeRedirect);
-                }
+
+              if (!promises.length) {
+                navigateToList();
+              } else {
+                // Wait until all promises resolve so that we can add alerts to
+                // AlertMessageService before navigating, otherwise they aren't
+                // displayed.
+                $q.all(promises).then(navigateToList);
```

Comment:
since we are waiting for a bunch of requests to complete now, are we disabling the Delete button after you hit it?  we should probably have been doing it anyway